config/errors.go

```diff
@@ -13,6 +15,83 @@ func (e ErrMapNotFound) Error() string {
 	return fmt.Sprintf("config: map (%v) not found", e.MapName)
 }
 
+type ErrParamTokenReserved struct {
+	MapName   string
+	Parameter provider.QueryParameter
+}
+
+func (e ErrParamTokenReserved) Error() string {
+	return fmt.Sprintf("config: map %s has parameter %s referencing reserved token %s",
+		e.MapName, e.Parameter.Name, e.Parameter.Token)
+}
+
+type ErrParamNameDuplicate struct {
+	MapName   string
+	Parameter provider.QueryParameter
+}
+
+func (e ErrParamNameDuplicate) Error() string {
+	return fmt.Sprintf("config: map %s redeclares duplicate parameter with name %s",
+		e.MapName, e.Parameter.Name)
+}
+
+type ErrParamTokenDuplicate struct {
+	MapName   string
+	Parameter provider.QueryParameter
+}
+
+func (e ErrParamTokenDuplicate) Error() string {
+	return fmt.Sprintf("config: map %s redeclares existing parameter token %s in param %s",
```

Comment:
```suggestion
	return fmt.Sprintf("config: map %s parameter %s has a token name %s already used by another parameter",
```
Also, let's be consistent - use either "parameter" or "param" (no preference here on my end).